core/base/inc/TVirtualQConnection.h

```diff
@@ -45,16 +45,6 @@ class TVirtualQConnection : public TList {
          SetArg((ULongptr_t)val);
    }
 
-   // We pass arrays as Lont_t*. In the template instance we can deduce their
-   // size, too.
-   template <class T, class = typename std::enable_if<std::is_array<T>::value>::type>
-   void SetArg(const T* val)
-   {
-      constexpr size_t size = sizeof(val)/sizeof(val[0]);
-      static_assert(size > 0, "The array must have at least one element!");
-      SetArg((Longptr_t*)val, size);
-   }
-
```

Comment:
@Axel-Naumann could you check my reasoning in the commit message? Am I missing something?

Reply:
It is likely the original author (@vgvassilev ?) meant:
```
   template <class T, class = typename std::enable_if<std::is_array<T>::value>::type>
   void SetArg(const T val)
```
Cheers,
Philippe.

Reply:
But @vgvassilev also says "We pass arrays as Lont_t*".

This is commit 95a27809bc55752d85a602e705d531b459de25b1 btw.

Reply:
Well .. I assumed the comment applied to the inner part ... but actually it hinted to the fact that in order to get the size of the array, we need to use a different syntax.

See: https://godbolt.org/z/xqo4ah61h   :   with `SetArg(const T val)` and `SetArg(arr1);` `T` is actually 'MyClass*`.

See: https://godbolt.org/z/4Yovca1Kr  :   with `SetArg(const T* val)` and `SetArg(&arr1);` then it works (and the code needs to be updated to de-reference the pointer)

See: https://godbolt.org/z/dj3K1MhbG  :   we can get `SetArg(arr1);`  to work with: `SetArg(const T (&val)[N])`

Reply:
Agreed, @hahnjo - this function has never been called. Tagging @bellenot so he's aware of this in case it ever bubbles up in signal/slot.